eigen.md

```diff
@@ -0,0 +1,293 @@
+                width=0.5))
+
+# Plot the lines they run through
+x = np.linspace(xmin, xmax, 3)
+for v in evecs:
+    a = v[1] / v[0]
+    ax.plot(x, a * x, 'b-', lw=0.4)
+
+plt.show()
+```
+
+The eigenvalue equation is equivalent to $(A - \lambda I) v = 0$, and
+this has a nonzero solution $v$ only when the columns of $A -
+\lambda I$ are linearly dependent.
+
+This in turn is equivalent to stating that the determinant is zero.
+
+Hence to find all eigenvalues, we can look for $\lambda$ such that the
+determinant of $A - \lambda I$ is zero.
+
+This problem can be expressed as one of solving for the roots of a polynomial
+in $\lambda$ of degree $n$.
+
+This in turn implies the existence of $n$ solutions in the complex
+plane, although some might be repeated.
+
+Some nice facts about the eigenvalues of a square matrix $A$ are as follows
+
+1. The determinant of $A$ equals  the product of the eigenvalues.
+1. The trace of $A$ (the sum of the elements on the principal diagonal) equals the sum of the eigenvalues.
+1. If $A$ is symmetric, then all of its eigenvalues are real.
+1. If $A$ is invertible and $\lambda_1, \ldots, \lambda_n$ are its eigenvalues, then the eigenvalues of $A^{-1}$ are $1/\lambda_1, \ldots, 1/\lambda_n$.
+
+A corollary of the first statement is that a matrix is invertible if and only if all its eigenvalues are nonzero.
+
+Using SciPy, we can solve for the eigenvalues and eigenvectors of a matrix as
+follows
+
+```{code-cell} python3
+A = ((1, 2),
+     (2, 1))
+
+A = np.array(A)
+evals, evecs = eig(A)
+evals
+```
+
+```{code-cell} python3
+evecs
+```
+
+Note that the *columns* of `evecs` are the eigenvectors.
+
+Since any scalar multiple of an eigenvector is an eigenvector with the same
+eigenvalue (check it), the eig routine normalizes the length of each eigenvector
+to one.
+
+
+
+
+### {index}`Positive Definite Matrices <single: Positive Definite Matrices>`
+
+```{index} single: Linear Algebra; Positive Definite Matrices
+```
+
+Let $A$ be a symmetric $n \times n$ matrix.
+
+We say that $A$ is
+
+1. *positive definite* if $x' A x > 0$ for every $x \in \mathbb R ^n \setminus \{0\}$
+1. *positive semi-definite* or *nonnegative definite* if $x' A x \geq 0$ for every $x \in \mathbb R ^n$
+
+Analogous definitions exist for negative definite and negative semi-definite matrices.
+
+It is notable that if $A$ is positive definite, then all of its eigenvalues
+are strictly positive, and hence $A$ is invertible (with positive
+definite inverse).
+
+
+NOTE Add examples, verify that results hold using NumPy
+
+
+
+## {index}`The Perron-Frobenius Theorem`
+
+In what follows, a matrix $A$ is called **nonnegative** if every element of $A$ is
+nonnegative.
+
+Often, in economics, the matrix that we are dealing with is nonnegative.
+
+Remarkably, as soon as we know that a matrix is nonnegative, we can also state
+that all of the eigenvalues and eigenvectors are real-valued rather than
+complex (i.e., they have no imaginary part).
```

Comment:
This is not quite true:

```py
A = [[0, 1, 0],
     [0, 0, 1],
     [1, 0, 0]]
np.linalg.eigvals(A)
```
```
array([-0.5+0.8660254j, -0.5-0.8660254j,  1. +0.j       ])
```

A nonnegative matrix has an eigenvalue that is real and attains the maximum absolute value among all eigenvalues (or equivalently, is equal to the spectral radius), and its corresponding eigenvector is (real- and) nonnegative-valued.

Reply:
Thanks @oyamad , that was a bad typo!

For context, these lectures are very rough ATM but we will be polishing them this coming week in a code / lecture writing sprint.

The idea is to have a set of lectures that can be taught to undergraduate students.

All further comments are appreciated.